src/screenComponents/databaseView.cpp

```diff
@@ -14,53 +14,45 @@ DatabaseViewComponent::DatabaseViewComponent(GuiContainer* owner)
 : GuiElement(owner, "DATABASE_VIEW")
 {
     database_entry = nullptr;
+    selected_entry = nullptr;
 
     item_list = new GuiListbox(this, "DATABASE_ITEM_LIST", [this](int index, string value) {
         P<ScienceDatabase> entry;
-        if (selected_entry)
-        {
-            if (index == 0)
-            {
-                selected_entry = selected_entry->parent;
-                fillListBox();
-            }else{
-                entry = selected_entry->items[index - 1];
-            }
-        }
-        else
-        {
-            entry = ScienceDatabase::science_databases[index];
-        }
-        display(entry);
+
+        unsigned int id = std::stoul(value, nullptr, 10);
+        selected_entry = findEntryById(id);
+        display();
     });
     item_list->setPosition(0, 0, ATopLeft)->setMargins(20, 20, 20, 130)->setSize(navigation_width, GuiElement::GuiSizeMax);
-    fillListBox();
+    display();
 }
 
-bool DatabaseViewComponent::findAndDisplayEntry(string name)
+P<ScienceDatabase> DatabaseViewComponent::findEntryById(unsigned int id)
 {
+    if (id == 0)
+    {
+        return nullptr;
+    }
     foreach(ScienceDatabase, sd, ScienceDatabase::science_databases)
     {
-        if (findAndDisplayEntry(name, sd))
-            return true;
+        if (sd->getId() == id)
+        {
+            return sd;
+        }
     }
-    return false;
+    return nullptr;
 }
 
-bool DatabaseViewComponent::findAndDisplayEntry(string name, P<ScienceDatabase> parent)
+bool DatabaseViewComponent::findAndDisplayEntry(string name)
 {
-    foreach(ScienceDatabase, sd, parent->items)
+    foreach(ScienceDatabase, sd, ScienceDatabase::science_databases)
```

Comment:
I know the foreach macro is used a lot in the code. But it can be replaced with range based for loops:
`for(auto sd : ScienceDatabase::science_databases)`
(Not critical, but if we are touching this line anyhow, we might as well fix that)

Reply:
Wow. This could become my new favorite syntax. :D

Reply:
"auto" and "range based for loops" are really powerful features of C++11. When I initially wrote EE, I didn't know about range based for loops, and that you could write your own custom iterators.